lib/actors/examples/chameneos-redux/chameneosredux.nit

```diff
@@ -0,0 +1,183 @@
+# This file is part of NIT (http://www.nitlanguage.org).
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Example implemented from "The computer Language Benchmarks Game" - Chameneos-Redux
+# http://benchmarksgame.alioth.debian.org/
+#
+# Complete description of the chameneos-redux :
+# https://benchmarksgame.alioth.debian.org/u64q/chameneosredux-description.html#chameneosredux
+module chameneosredux is no_warning("missing-doc")
+
+import actors
+
+class Creature
+	actor
+	var place: MeetingPlace
+	var color: Int
+	var id: Int
+	var count = 0
+	var samecount = 0
+
+	fun run do
+		loop
+			var p = place.meet(id, color)
+			if p == null then break
+			color = p.color
+			if p.sameid then samecount += 1
+			count += 1
+		end
+	end
+
+	fun to_string: String do return count.to_s + " " + numbers[samecount]
```

Comment:
I wish... At this point my generation phase is not really excluding methods like `to_s` from being proxied. If I redefine it, I get an error because in the proxy class my generation would try to change the signature to `fun to_s: Future[String]`. Even before that, the error would be that I'm trying to define an inherited property without redef keyword in my proxy class.